### PR TITLE
adding random passphrase generation on new created surveyresults

### PR DESCRIPTION
**Pull request by qwo**

### CODE REVIEW

**Comment by rydente** [comment]
Tests?

Also, conflicts for now.

File: api/controllers/SurveyController.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"use strict";
+
+const Controller = require("trails-controller");
+const Boom = require("boom");
+const words = require("./words.json");
+const _ = require("lodash");
+module.exports = class SurveyResultController extends Controller {
```
Comment (ttavenner) [inline api/controllers/SurveyController.js]: This should be in SuveyResultController.

File: api/controllers/index.js [hunk 2]
```diff
@@ -6,3 +6,4 @@ exports.AuthController = require('./AuthController');
 exports.UserController = require('./UserController');
 exports.CategoryController = require('./CategoryController');
 exports.QuestionController = require('./QuestionController');
+exports.SurveyResultController = require('./SurveyController');
```
Comment (ttavenner) [inline api/controllers/index.js]: Should point at SurveyResultController

File: api/services/SurveyResultService.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+"use strict";
```
Comment (ttavenner) [inline api/services/SurveyResultService.js]: Make sure you have pulled in the most recent changes, this file already exists and has several functions.

File: api/services/index.js [hunk 4]
```diff
@@ -5,3 +5,5 @@ exports.AuthService = require('./AuthService');
 exports.UserService = require('./UserService');
 exports.CategoryService = require('./CategoryService');
 exports.QuestionService = require('./QuestionService');
+exports.SurveyResultService = require('./SurveyResultService');
```
Comment (ttavenner) [inline api/services/index.js]: Make sure you have the most recent changes, this line already exists as well as another line.

File: config/routes.js [hunk 5]
```diff
@@ -99,5 +99,10 @@ module.exports = [
         method: ['GET'],
         path: '/api/v1/default/info',
         handler: 'DefaultController.info'
+    },
+    {
+        method: 'POST',
+        path: '/api/v1/survey',
```
Comment (ttavenner) [inline config/routes.js]: This should be on the `surveyresult` route rather than the `survey` route.

**Review by ttavenner — CHANGES_REQUESTED** [review]

File: api/controllers/SurveyController.js (hunk 1 above)
Comment (qwo) [reply api/controllers/SurveyController.js]: 👍

File: api/controllers/index.js (hunk 2 above)
Comment (qwo) [reply api/controllers/index.js]: 👍

**Comment by qwo** [comment]
@rydente  good point will add tests

@ttavenner merged up stream thanks for that hint. 

There are also a few broken jshint rules that ill need to also patch.

**Comment by qwo** [comment]
came back and added a test here @rydente 

https://github.com/Code4HR/okcandidate-platform/pull/141/commits/3edd36f086f1eabcd6a441d5953afeaf6db077c9

otherwise nothing too much changed.

**Comment by qwo** [comment]
addressed all your changes @ttavenner :)

**Comment by ttavenner** [comment]
The API tests currently fails when running `npm test` because it expects the API to already be running.

**Comment by qwo** [comment]
this doesn't fail for me locally. Does it fail on codeship @ttavenner ? How can i have the build run manually curious don't see it.

**Comment by ttavenner** [comment]
There is no way to run it manually on Codeship @stanzheng, it won't build off of forks. However since it is running the test using `npm test` in a linux container there is a high likelihood it will have the same result as my environment. Are you running at `npm test` with the server running in the background? If I do that it passes but if I run `npm test` without starting the server it fails with the following error:

```
SurveyResultController generate a public and private passphrase:
     TypeError: Cannot read property 'status' of undefined
      at Test._assertStatus (node_modules/supertest/lib/test.js:229:10)
      at Test._assertFunction (node_modules/supertest/lib/test.js:247:11)
      at Test.assert (node_modules/supertest/lib/test.js:148:18)
      at assert (node_modules/supertest/lib/test.js:127:12)
      at node_modules/supertest/lib/test.js:124:5
      at Test.Request.callback (node_modules/superagent/lib/node/index.js:687:12)
      at ClientRequest.<anonymous> (node_modules/superagent/lib/node/index.js:639:10)
      at Socket.socketErrorListener (_http_client.js:309:9)
      at emitErrorNT (net.js:1281:8)
      at _combinedTickCallback (internal/process/next_tick.js:80:11)
      at process._tickDomainCallback (internal/process/next_tick.js:128:9)
```

**Comment by qwo** [comment]
@ttavenner so i dunno, it seems like on first run, while the service is starting up its down but waiting a while it actually hits the  live service. 

Probably the wrong way to test this. I can remove it to get the pull in. 

@ryayak1460 @rydente might have a better suggestion how to test this.

**Comment by rydente** [comment]
If you really want to run an integration test, you'll need to start the service and wait til it starts in the `npm test` command.

Otherwise, I'd suggest writing unit tests instead.  They won't require the service to run at all.  You could mock dependencies with a library, possibly [Sinon](http://sinonjs.org/).

**Comment by qwo** [comment]
ok test fixed @wbprice mergy mergy.

**Comment by qwo** [comment]
the failed build is from the HR file change. thats weird.

**Comment by wbprice** [comment]
I don't know.  It could be some kind of caching in CI.  @ttavenner

**Pull request closed without merging**